docs/programming_guide.md

```diff
@@ -2398,6 +2399,61 @@ of a function, or use the `withParameters(...)` method to pass in a configuratio
 [Back to top](#top)
 
 
+Passing parameters to functions
+-------------------
+
+Parameters can be passed to rich functions using either the constructor (if the function is
```

Comment:
If the function is not serialisable the program will fail in any case. That is not related to the configuration. 
IMO, the documentation should state that the function object is serialised using standard java serialization and shipped "as-is" to all parallel task instances. Mentioning that transient fields are not shipped is OK but not mandatory (somebody who uses the transient keyword should know what it does).